<?xml version="1.0"?>
<launch>
  <arg name="paused" default="false"/>
  <arg name="use_sim_time" default="true"/>
  <arg name="gui" default="true"/>
  <arg name="headless" default="false"/>
  <arg name="debug" default="false"/>
  <arg name="verbose" default="false"/>
  <arg name="world_name" default="4cubes"/>
  <arg name="x_pos" default="0" />
  <arg name="y_pos" default="0" />
  <arg name="z_pos" default="0"/>
  <include file="$(find gazebo_ros)/launch/empty_world.launch" >
    <arg name="world_name" value="$(find sim_description)/worlds/$(arg world_name).world"/>
    <arg name="paused" value="$(arg paused)"/>
    <arg name="use_sim_time" value="$(arg use_sim_time)"/>
    <arg name="gui" value="$(arg gui)"/>
    <arg name="headless" value="$(arg headless)"/>
    <arg name="debug" value="$(arg debug)"/>
    <arg name="verbose" value="$(arg verbose)"/>

  </include>

  <arg name="gpu" default="false"/>
<param name="robot_description" command="$(find xacro)/xacro --inorder '$(find sim_description)/urdf/sim.urdf.xacro' gpu:=$(arg gpu)" />
<node pkg="gazebo_ros" type="spawn_model" name="spawn_model" args="-urdf -x $(arg x_pos) -y $(arg y_pos) -z $(arg z_pos) -param /robot_description -model lidar_sim"/> -->
<node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
  <param name="publish_frequency" type="double" value="30.0" />
</node>
<node name="tf_publisher" pkg="sim_description" type="tf_transform.py" output="screen"/>
<node name="lidar_tf_publisher" pkg="sim_description" type="tf_broadcaster" output="screen"/>
<node name="odom_publisher" pkg="sim_description" type="odom_pub.py" output="screen"/>
<!-- <node name="lidar_Caspr" pkg="sim_description" type="lidar_pub.py" output="screen"/> -->

</launch>
